src/core/syscalls/os_syscall_handler.rs

```diff
@@ -1,2 +1,944 @@
+            VecDeque::new(),
+            HashMap::new(),
+            Some(Relocatable {
+                segment_index: 0,
+                offset: 0,
+            }),
+            None,
+        );
+        let get_contract_address = handler._get_contract_address();
+        assert_eq!(get_contract_address, Ok(5))
+    }
+
+    #[test]
+    fn get_contract_address_err() {
+        let handler = OsSyscallHandler::new(
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            HashMap::new(),
+            Some(Relocatable {
+                segment_index: 0,
+                offset: 0,
+            }),
+            None,
+        );
+        let get_contract_address = handler._get_contract_address().unwrap_err();
+        assert_eq!(get_contract_address, SyscallHandlerError::ListIsEmpty)
+    }
+
+    #[test]
+    fn end_tx_err_execute_code_read_iterator() {
+        let mut execute_code_read_iterator = VecDeque::new();
+        execute_code_read_iterator.push_back(12);
+        let mut handler = OsSyscallHandler::new(
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            execute_code_read_iterator,
+            HashMap::new(),
+            None,
+            None,
+        );
+
+        assert_eq!(handler.end_tx(), Err(SyscallHandlerError::IteratorNotEmpty))
+    }
+    #[test]
+    fn end_tx_err_tx_info_ptr() {
+        let tx_info_ptr = Some(Relocatable {
+            segment_index: 0,
+            offset: 0,
+        });
+        let mut handler = OsSyscallHandler::new(
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            HashMap::new(),
+            tx_info_ptr,
+            None,
+        );
+
+        assert_eq!(
+            handler.end_tx(),
+            Err(SyscallHandlerError::ShouldBeNone(String::from(
+                "tx_info_ptr"
+            )))
+        )
+    }
+
+    #[test]
+    fn end_tx_err_tx_execution_info() {
+        let mut handler = OsSyscallHandler::new(
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            HashMap::new(),
+            None,
+            Some(TransactionExecutionInfo),
+        );
+
+        assert_eq!(
+            handler.end_tx(),
+            Err(SyscallHandlerError::ShouldBeNone(String::from(
+                "tx_execution_info",
+            )))
+        )
+    }
+
+    #[test]
+    fn end_tx() {
+        let mut handler = OsSyscallHandler::new(
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
```

Comment:
A macro could be use to create the handler as this is a recurrent pattern in the tests.